File: docs/examples/node/embed.js
// examples/node/embed.js — get a text embedding via JumpNet.
// Usage:  node embed.js "your text here"

const JUMPNET = process.env.JUMPNET_URL ?? 'http://localhost:4080';

async function embed(text, dimensions = 128) {
  const res = await fetch(`${JUMPNET}/embed`, {
    method:  'POST',
    headers: { 'Content-Type': 'application/json' },
    body:    JSON.stringify({ text, dimensions }),
  });
  if (!res.ok) throw new Error(`HTTP ${res.status}`);
  return res.json();
}

const text = process.argv.slice(2).join(' ') || 'round red bead';
embed(text)
  .then(r => {
    console.log(`Model:      ${r.model}`);
    console.log(`Dimensions: ${r.dimensions}`);
    console.log(`First 8:    ${r.embedding.slice(0, 8).join(', ')}`);
  })
  .catch(e => { console.error(e.message); process.exit(1); });

File: docs/examples/python/embed.py
"""
examples/python/embed.py — generate a text embedding via JumpNet.
Usage:  python embed.py "round red bead"
"""
import sys, json, urllib.request

JUMPNET = "http://localhost:4080"

def embed(text: str, dimensions: int = 128) -> dict:
    payload = json.dumps({"text": text, "dimensions": dimensions}).encode()
    req = urllib.request.Request(
        f"{JUMPNET}/embed",
        data=payload,
        headers={"Content-Type": "application/json"},
        method="POST",
    )
    with urllib.request.urlopen(req) as resp:
        return json.loads(resp.read())

if __name__ == "__main__":
    text = " ".join(sys.argv[1:]) or "round red bead"
    result = embed(text)
    print(f"Model:      {result['model']}")
    print(f"Dimensions: {result['dimensions']}")
    print(f"First 8:    {result['embedding'][:8]}")
